packages/fxa-auth-server/lib/routes/subscriptions.js

```diff
@@ -471,32 +472,163 @@ class DirectStripeRoutes {
     return response;
   }
 
+  /**
+   * Send a subscription status Service Notification event to SQS
+   *
+   * @param {*} request
+   * @param {string} uid
+   * @param {Event} event
+   * @param {{id: string, productId: string}} sub
+   * @param {boolean} isActive
+   */
+  async sendSubscriptionStatusToSqs(request, uid, event, sub, isActive) {
+    this.log.notifyAttachedServices('subscription:update', request, {
+      uid,
+      eventCreatedAt: event.created,
+      subscriptionId: sub.id,
+      isActive,
+      productId: sub.productId,
+      productCapabilities:
+        this.config.subscriptions.productCapabilities[sub.productId] || [],
+    });
+  }
+
+  /**
+   * Handle `subscription.created` Stripe webhook events.
+   *
+   * Only subscriptions that are active/trialing are valid. Emit an event for
+   * those conditions only.
+   *
+   * @param {*} request
+   * @param {Event} event
+   */
+  async handleSubscriptionCreatedEvent(request, event) {
+    const sub = /** @type {Subscription} */ (event.data.object);
+    if (!['active', 'trialing'].includes(sub.status)) {
+      return;
+    }
+    const { uid, email } = await this.getCustomerUidEmailFromSubscription(sub);
+    if (!uid) {
+      return;
+    }
+    await this.stripeHelper.refreshCachedCustomer(uid, email);
+    await this.profile.deleteCache(uid);
+    await this.sendSubscriptionStatusToSqs(
+      request,
+      uid,
+      event,
+      { id: sub.id, productId: /** @type {string} */ (sub.plan.product) },
+      true
+    );
```

Comment:
Nit: Might be worth extracting this chunk as a method shared between the handlers? Seems like the same code for each, except for the logic in `handleSubscriptionUpdatedEvent` which could be an early return like here in this handler

Reply:
Sort of, the order is different so that some work can be avoided in the updated event handler, which means only the deleted also would benefit from that. But that might still be cleaner.